### PR TITLE
XFail any test that uses pfc_gen.py

### PR DESCRIPTION
**Pull request by rraghav-cisco**
### Description of PR
The script pfc_gen.py that is used to produce pfc xoff from fanout host is not as reliable as we hope it to be. There are many instances where the xoff is sent with a larger gap than needed, and the DUT ends up leaking out the packets in that interval. This causes many pfc tests to fail. All these tests have to be re-written with ixia or spirent.

### Type of change

- [X] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [] 201911
- [X] 202012
- [X] 202205

#### What is the motivation for this PR?
Until we have a set of new tests that can replace the current set of pfcwd tests, we have to xfail them.

#### How did you do it?
I have set the tests as xfail in the mark file.

#### How did you verify/test it?
I ran one of the tests, and it came out xpass when it passed:
----------------------------------------------------------------------------------------------------- live log teardown ------------------------------------------------------------------------------------------------------
18:45:54 conftest.core_dump_and_config_check      L1757 WARNING| Core dump or config check failed for test_pfcwd_timer_accuracy.py, results: {"core_dump_check": {"new_core_dumps": {"mth64-m5-2": []}, "pass": true}, "config_db_check": {"cur_only_config": {"mth64-m5-2": {"PFC_WD": {"Ethernet180": {"action": "drop", "detection_time": "400", "restoration_time": "400"}, "GLOBAL": {"POLL_INTERVAL": "400"}}}}, "inconsistent_config": {"mth64-m5-2": {}}, "pre_only_config": {"mth64-m5-2": {}}, "pass": false}}

-------------------------------------------------------------- generated xml file: /home/vxr/vxlan/logs/pfcwd/test_pfcwd_timer_accuracy_2023-02-09-18-35-51.xml --------------------------------------------------------------
================================================================================================== short test summary info ===================================================================================================
XPASS pfcwd/test_pfcwd_timer_accuracy.py::TestPfcwdAllTimer::test_pfcwd_timer_accuracy[None] The fanout script(pfc_gen.py) is not reliable.
================================================================================================ 1 xpassed in 893.49 seconds =================================================================================================
INFO:root:Can not get Allure report URL. Please check logs
vxr@02b4831febde:/data/tests$

### CODE REVIEW

**Comment by rraghav-cisco** [comment]
@kevinskwang , @neethajohn : Pls review.

File: tests/common/plugins/conditional_mark/tests_mark_conditions.yaml [hunk 1]
```diff
@@ -611,6 +611,22 @@ pfcwd/test_pfc_config.py::TestPfcConfig::test_forward_action_cfg:
      conditions:
         - "asic_type in ['cisco-8000']"
 
+pfcwd/test_pfcwd_warm_reboot.py:
```
Comment (neethajohn) [inline tests/common/plugins/conditional_mark/tests_mark_conditions.yaml:614]: Please make all these changes specific to cisco platform for the time being

Comment (rraghav-cisco) [reply tests/common/plugins/conditional_mark/tests_mark_conditions.yaml:614]: @neethajohn , Ack. I will change them to cisco-8000 only.

Comment (neethajohn) [reply tests/common/plugins/conditional_mark/tests_mark_conditions.yaml:614]: @rraghav-cisco, is there an ETA on when the replacement testcases will be available? These are really important testcases and not having them covered is a major test gap

Comment (rraghav-cisco) [reply tests/common/plugins/conditional_mark/tests_mark_conditions.yaml:614]: @neethajohn , We are looking at the ixia tests for comparison. We are yet to complete the comparison.

From my preliminary check, the following tests are not covered. The rest are covered.
1. pfcwd/test_pfcwd_timer_accuracy.py,
2. pfcwd/test_pfcwd_all_port_storm.py and
3. qos_sai testcases.

**Comment by kevinskwang** [comment]
> @kevinskwang , @neethajohn : Pls review.

@rraghav-cisco I saw all those cases are consistent passed on 8102. does it really make sense to xfail on non 400G platform?

**Comment by rraghav-cisco** [comment]
> 

@kevinskwang , even if they pass in 100G, there is always a chance to fail. That is one reason the PR is xfailing it instead of skipping them.

File: tests/common/plugins/conditional_mark/tests_mark_conditions.yaml (hunk 1 above)
Comment (rraghav-cisco) [reply tests/common/plugins/conditional_mark/tests_mark_conditions.yaml:614]: @neethajohn,
We plan to provide a PR for all-port storm : 3/31.
timer-accuracy: Has dependency on ixia, hope to pick it up on 4/7.

Also, until the new PRs are merged, can we keep the xfail ? Once the new PRs are merged, we can mark the corresponding tests as skipped.

**Comment by rlhui** [comment]
this is also applicable to t2 right?

**Comment by rraghav-cisco** [comment]
@rlhui Yes, this is applicable to any topology.

**Comment by kevinskwang** [comment]
Close as discussed, we will continue to run it.

**Pull request closed without merging**